### PR TITLE
Issue 20060 wf multiple contentlet

### LINKED ISSUE #20060: Allow multiple contents to be put via API
dotCMS has REST apis for saving content in the system, most notably, the content and workflowAPIs.  This allows for single pieces of content to be put one at a time in dotCMS.

In addition to these endpoints, we need another endpoint that can take multiple contents/assets.  The format for these puts should be similar to our current workflow API put, where you can specify a workflow and a contentlet as a json object - but instead it should take a workflow and then array of content objects and loop over them (perhaps multithreaded?) and ingest them in dotCMS.   Each content object created would get pushed through the same workflow step/process.

If we mutlithreaded, we should 1. make that configurable and 2. start with say 4 threads.

### PR DESCRIPTION
**Pull request by jdotcms**
First intent of the fire multiple contentlets in just one request

### CODE REVIEW

**Comment by erickgonzalez** [comment]
Tests related to Workflow Resource are failing.

File: dotCMS/src/main/java/com/dotcms/workflow/form/FireMultipleActionForm.java [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+import com.dotcms.rest.api.Validated;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
+import java.util.List;
+import java.util.Map;
+
+@JsonDeserialize(builder = FireMultipleActionForm.Builder.class)
+public class FireMultipleActionForm extends Validated {
```
Comment (jcastro-dotcms) [inline dotCMS/src/main/java/com/dotcms/workflow/form/FireMultipleActionForm.java:11]: Javadoc

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/WorkflowResource.java [hunk 2]
```diff
@@ -1576,6 +1577,134 @@ public final Response fireActionDefault(@Context final HttpServletRequest reques
...
+        @Override
+        public void write(final OutputStream output) throws IOException, WebApplicationException {
+
+            final ObjectMapper objectMapper = DotObjectMapperProvider.getInstance().getDefaultObjectMapper();
+            WorkflowResource.this.saveMultipleContentletsByDefaultAction(this.contentletsToSaveList, this.systemAction, this.fireActionForm,
+                    this.request, this.mode, this.initDataObject, output, objectMapper);
+        }
+    }
+
+    private void saveMultipleContentletsByDefaultAction(final List<Map<String, Object>> contentletsToSaveList,
```
Comment (jcastro-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/WorkflowResource.java:1647]: Javadoc

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/WorkflowResource.java [hunk 3]
```diff
@@ -1576,6 +1577,134 @@ public final Response fireActionDefault(@Context final HttpServletRequest reques
...
+                                singleFireActionForm, initDataObject, mode);
+
+                try {
+
+                    fireTransactionalAction(systemAction, singleFireActionForm, request, mode,
+                            initDataObject, resultMap, inode, identifier, languageId, user, indexPolicy);
+                } catch (Exception e) {
+
+                    final String id = UtilMethods.isSet(identifier)?identifier:inode;
+                    Logger.error(this, "Error in contentlet: " + id + ", msg: " + e.getMessage(), e);
```
Comment (jcastro-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/WorkflowResource.java:1692]: Is it possible to include what specific action failed for this content? If it is, we need to log it.

**Review by jcastro-dotcms — CHANGES_REQUESTED** [review]

**Review by erickgonzalez — APPROVED** [review]

**Review by jcastro-dotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/WorkflowResource.java [hunk 4]
```diff
@@ -1576,6 +1578,146 @@ public final Response fireActionDefault(@Context final HttpServletRequest reques
...
+     * @param response   {@link HttpServletResponse}
+     * @param fireActionForm {@link FireActionForm} Fire Action Form
+     * @param systemAction {@link com.dotmarketing.portlets.workflows.business.WorkflowAPI.SystemAction} system action to determine the default action
+     * @return Response
+     */
+    @POST()
+    @Path("/actions/default/fire/{systemAction}")
+    @JSONP
+    @NoCache
+    //@Produces({MediaType.APPLICATION_JSON, "application/javascript"})
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/WorkflowResource.java:1597]: Please remove this commented line

**Review by nollymar — APPROVED** [review]

**Pull request merged**